rust/operator-binary/src/main.rs

```diff
@@ -46,7 +47,10 @@ async fn main() -> anyhow::Result<()> {
             ))
             .await?;
 
-            restart_controller::start(&client).await?
+            let sts_restart_controller = restart_controller::statefulset::start(&client);
+            let pod_restart_controller = restart_controller::pod::start(&client);
+            pin_mut!(sts_restart_controller, pod_restart_controller);
+            futures::future::select(sts_restart_controller, pod_restart_controller).await;
```

Comment:
Looking into the docs https://docs.rs/futures/latest/futures/future/fn.select.html `Waits for either one of two differently-typed futures to complete.` im a little bit confused. Shouldn't this wait for both futures to finish with something like `select_all`?

Reply:
`select` and `select_all` implement the same basic semantics, with the difference being like that of tuples and vecs:

1. `select` is hard-coded to work with exactly two futures (although multiple `select` calls can be chained if you know all cases statically), `select_all` takes an arbitrarily long `Vec`
2. `select` supports heterogenous types, `select_all` requires all futures to have the same type (or that they're boxed into submission)
3. `select` is a bit more transparent to the rustc/llvm optimizer, since we keep more information about the underlying types
4. `select` doesn't require an allocation, `select_all` requires at least one for the underlying `Vec`, plus usually one allocation per `Future` for the `Box`ening

Reply:
Thanks a lot!